olmo/optim.py

```diff
@@ -647,6 +649,177 @@ def get_post_step_metrics(
             return metrics
 
 
+class DeMo(torch.optim.SGD, Optimizer):
```

Comment:
It seems like the organization would make more sense if this class, and `demo_utils.py`, were in their own file together, and then we use `__all__` to make this optimizer appear the same as the others.